test_fatoptimizer.py

```diff
@@ -2829,6 +2829,19 @@ def f(x):
                 return 42 + 3
         ''')
 
+    def test_nested_function(self):
+        self.check_optimize('''
+            def f(x):
+                def g(x):
+                    return 100
+                return g(x) + 3
```

Comment:
I recently disabled specialization of nested functions because it killed performances. If the nested function is only called once, it looks like the specialization (creation of guards, call to fat.specialize()) is more expensive than optimizations implemented in FAT Python.

Currently, there is no heuristic to decide if it's worth to specialize a function or not.